### ISSUE TITLE
TestGolden{Add,Init}Cmd time-dependent thus failing reproducibility test

### ISSUE THREAD

**Issue by anthonyfok** [spf13/cobra#503]
TestGoldenAddCmd and TestGoldenInitCmd compares newly generated files containing current year with pre-generated "golden" files set in 2017.  This fails "Reproducible builds" as explained in https://reproducible-builds.org/.

From https://tests.reproducible-builds.org/debian/logs/unstable/arm64/golang-github-spf13-cobra_0.0~git20170724.0.90fc11b-3.build2.log.gz (transient URL):

(or https://web.archive.org/web/20170729172939/https://tests.reproducible-builds.org/debian/logs/unstable/arm64/golang-github-spf13-cobra_0.0~git20170724.0.90fc11b-3.build2.log.gz for a permanent URL):

```
I: Current time: Sat Sep  1 09:01:29 +14 2018

...

dpkg-source: info: extracting golang-github-spf13-cobra in 2nd
dpkg-source: info: unpacking golang-github-spf13-cobra_0.0~git20170724.0.90fc11b.orig.tar.xz
dpkg-source: info: unpacking golang-github-spf13-cobra_0.0~git20170724.0.90fc11b-3.debian.tar.xz
dpkg-source: info: applying 001-fix-TestGlobalNormFuncPropagation-on-gccgo.patch

...

   dh_auto_test -O--buildsystem=golang
	go test -v -p 8 github.com/spf13/cobra github.com/spf13/cobra/cobra github.com/spf13/cobra/cobra/cmd github.com/spf13/cobra/doc

...

=== RUN   TestGoldenAddCmd
--- FAIL: TestGoldenAddCmd (0.01s)
	add_test.go:69: "/build/golang-github-spf13-cobra-0.0~git20170724.0.90fc11b/2nd/obj-aarch64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go" and "testdata/root.go.golden" are not equal!
		
		$ diff /build/golang-github-spf13-cobra-0.0~git20170724.0.90fc11b/2nd/obj-aarch64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go testdata/root.go.golden
		1c1
		< // Copyright © 2018 NAME HERE <EMAIL ADDRESS>
		---
		> // Copyright © 2017 NAME HERE <EMAIL ADDRESS>
		
		exit status 1
=== RUN   TestValidateCmdName
--- PASS: TestValidateCmdName (0.00s)
=== RUN   TestGoldenInitCmd
--- FAIL: TestGoldenInitCmd (0.01s)
	init_test.go:67: "/build/golang-github-spf13-cobra-0.0~git20170724.0.90fc11b/2nd/obj-aarch64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go" and "testdata/root.go.golden" are not equal!
		
		$ diff /build/golang-github-spf13-cobra-0.0~git20170724.0.90fc11b/2nd/obj-aarch64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go testdata/root.go.golden
		1c1
		< // Copyright © 2018 NAME HERE <EMAIL ADDRESS>
		---
		> // Copyright © 2017 NAME HERE <EMAIL ADDRESS>
		
		exit status 1
=== RUN   TestFindExistingPackage
--- PASS: TestFindExistingPackage (0.00s)
FAIL
FAIL	github.com/spf13/cobra/cobra/cmd	0.052s
```

I wonder how to keep these tests while satisfying reproducibility?
Would some kind of fake time.Now work?

**Comment by anthonyfok** [comment]
Perhaps using `SOURCE_DATE_EPOCH` as proposed at
https://wiki.debian.org/ReproducibleBuilds/TimestampsProposal
would work?

Example for Go:
(from https://wiki.debian.org/ReproducibleBuilds/TimestampsProposal#Go)

```go
import (
        "fmt"
        "os"
        "strconv"
        "time"
)

[...]

source_date_epoch := os.Getenv("SOURCE_DATE_EPOCH")
var build_date string
if source_date_epoch == "" {
        build_date = time.Now().UTC().Format(http.TimeFormat)
} else {
        sde, err := strconv.ParseInt(source_date_epoch, 10, 64)
        if err != nil {
                panic(fmt.Sprintf("Invalid SOURCE_DATE_EPOCH: %s", err))
        }
        build_date = time.Unix(sde, 0).UTC().Format(http.TimeFormat)
}
```

Or maybe not?  (It is not a build timestamp after all).

Maybe this?  https://play.golang.org/p/_AMjbv2qWB

Or maybe this?  http://godoc.org/github.com/stephanos/clock

(Sorry, just thinking out loud, doing a brain dump, etc.)

**Comment by anthonyfok** [comment]
Oops, my fix in #504 is incomplete, as it is still failing on the "future date in 2018" reproducible builds test machine:

```
--- FAIL: TestGoldenAddCmd (0.01s)
	add_test.go:74: "/build/golang-github-spf13-cobra-0.0~git20170730.0.d7ba195/2nd/obj-x86_64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go" and "testdata/root.go.golden" are not equal!
		
		$ diff -u /build/golang-github-spf13-cobra-0.0~git20170730.0.d7ba195/2nd/obj-x86_64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go testdata/root.go.golden
		--- /build/golang-github-spf13-cobra-0.0~git20170730.0.d7ba195/2nd/obj-x86_64-linux-gnu/src/github.com/spf13/testproject/cmd/root.go	2018-09-03 11:42:14.671657005 +1400
		+++ testdata/root.go.golden	2018-09-03 11:42:08.551657005 +1400
		@@ -1,4 +1,4 @@
		-// Copyright © 2018 NAME HERE <EMAIL ADDRESS>
		+// Copyright © 2017 NAME HERE <EMAIL ADDRESS>
		 //
		 // Licensed under the Apache License, Version 2.0 (the "License");
		 // you may not use this file except in compliance with the License.
		
		exit status 1
```

Full log at https://web.archive.org/web/20170731161924/https://tests.reproducible-builds.org/debian/logs/unstable/amd64/golang-github-spf13-cobra_0.0~git20170730.0.d7ba195-1.build2.log.gz

It turns out that I did the "tweaking" of the copyright year a little bit too late.  It should have been before `initializeProject(project)`, not after.  Sorry for my oversight.  (Brown paper bag over my head.)  :wink:

Will fix this in the next commit.
